mesonbuild/modules/pkgconfig.py

```diff
@@ -526,7 +526,7 @@ def parse_variable_list(stringlist):
             for lib in deps.pub_libs:
                 if not isinstance(lib, str) and not hasattr(lib, 'generated_pc'):
                     lib.generated_pc = filebase
-                    location = types.SimpleNamespace(subdir=state.subdir,
+                    location = types.SimpleNamespace(filename=os.path.join(state.subdir, state.build_filename),
```

Comment:
Oh, if we already have the relevant node, this can just be `location = self.current_node`

Reply:
I'm not sure to follow you, `current_node` is not defined

Reply:
Bah, I mean `location = state.current_node` it's an attribute of the ModuleState object.

Reply:
even better indeed.